game-core/src/main/java/games/strategy/triplea/delegate/TransportTracker.java

```diff
@@ -139,17 +129,10 @@ public static Change unloadAirTransportChange(
       return change;
     }
     assertTransport(transport);
-    if (!transport.getTransporting().contains(unit)) {
-      throw new IllegalStateException(
-          "Not being carried, unit:" + unit + " transport:" + transport);
```

Comment:
This is the exception being thrown.  I removed it because it really is red-herring.  The code `transport.getTransporting()` goes through all the territories on the map to find the one territory that has the transport.  It then loops through all of the units in that territory and finds all the units that have `Unit#getTransportedBy` equal to the transport.  And then, this method just checks if the `unit` is in that list.

But if you look a few lines up (line 127), it grabbed the transport by calling `Unit#getTransportedBy`.  So `transport.getTransporting().contains(unit)` will always return true except for the odd situation of when the unit and its transporter are in different territories.  And in this case, the unit wasn't in any territory because it had been removed.

This check was originally added when transports kept a list of the units they were transporting and back then this check made sense.  But transports have changed and it now is redundant.